src/libtiled/orthogonalrenderer.cpp

```diff
@@ -225,10 +270,11 @@ void OrthogonalRenderer::drawTileLayer(QPainter *painter,
     const QPointF layerPos(layer->x() * tileWidth,
                            layer->y() * tileHeight);
 
-    int startX = 0;
-    int startY = 0;
-    int endX = layer->width() - 1;
-    int endY = layer->height() - 1;
+    QRect bounds = layer->bounds().translated(-layer->position());
+    int startX = bounds.left();
+    int startY = bounds.top();
+    int endX = bounds.right();
+    int endY = bounds.bottom();
```

Comment:
It was previously iterating upto `tileLayer->width() - 1`. So if `- 1` is added, then it doesn't render the last row/column.